src/apps/velodrome/optimism/velodrome.bribe.contract-position-fetcher.ts

```diff
@@ -0,0 +1,101 @@
+import axios from 'axios';
+import { Inject } from '@nestjs/common';
+import { BigNumberish } from 'ethers/lib/ethers';
+import { MetaType } from '~position/position.interface';
+import { range } from 'lodash';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import {
+    DefaultContractPositionDefinition,
+    GetDisplayPropsParams,
+    GetTokenBalancesParams,
+} from '~position/template/contract-position.template.types';
+
+import { VelodromeContractFactory, VelodromeBribe } from '../contracts';
+
+interface VelodromeApiPairData {
+    gauge: GaugeData;
+}
+
+interface GaugeData {
+    wrapped_bribe_address: string;
+}
+
+@PositionTemplate()
+export class OptimismVelodromeBribeContractPositionFetcher extends ContractPositionTemplatePositionFetcher<VelodromeBribe> {
+    constructor(
+        @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+        @Inject(VelodromeContractFactory) protected readonly contractFactory: VelodromeContractFactory,
+    ) {
+        super(appToolkit);
+    }
+    veTokenAddress = '0x9c7305eb78a432ced5c4d14cac27e8ed569a2e26';
+    veVoteAddress = '0x09236cff45047dbee6b921e00704bed6d6b8cf7e';
+    groupLabel = 'Bribe';
+
+    getContract(address: string): VelodromeBribe {
+        return this.contractFactory.velodromeBribe({ address, network: this.network });
+    }
+
+    async getDefinitions(): Promise<DefaultContractPositionDefinition[]> {
+        const { data } = await axios.get<{ data: VelodromeApiPairData[] }>('https://api.velodrome.finance/api/v1/pairs');
+        var definitions = new Array;
+        data.data.filter(v => !!v).filter(v => !!v.gauge).forEach(pool => {
+            const wBribeAddress = pool.gauge.wrapped_bribe_address;
+            if (wBribeAddress != null) {
+                definitions.push({ address: wBribeAddress });
+            }
+        }
+        );
+        return definitions;
+    }
+
+    async getTokenDefinitions({ contract }) {
+        const numRewards = Number(await contract.rewardsListLength());
+        const bribeTokens = await Promise.all(range(numRewards).map(async n => await contract.rewards(n)));
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(this.network);
```

Comment:
You don't need to deal/filter out with baseTokens directly, you can do something like 
```
return bribeTokens.map(bribeToken => ({
        metaType: MetaType.CLAIMABLE,
        address: bribeToken.address,
        network: this.network,
    })
)
```

I'm guessing you used `await this.appToolkit.getBaseTokenPrices(this.network);` because some position weren't returned right ?

Reply:
I started with what you suggest but any position with an unknown token was removed from the list at the "skeletonsWithResolvedTokens" stage of getPositions token.
That's almost half of the bribed pool, mostly due to OptiDoge.

Reply:
Ohh I see.. What do you think about filtering out undesirable tokens like OptiDoge ?

Reply:
I tried filtering keeping only known base tokens, maybe this could be enlarged to app tokens as well.
Seems easier to filter using a token whitelist than a blacklist.

Reply:
You can implement either solution, both are fine.

Reply:
Do you see an issue merging it as it is now? Then maybe update filter if too many app tokens are getting removed

Reply:
I'll merge it and I'll make smol changes. Well done !